gossipd/gossip.c

```diff
@@ -358,11 +357,10 @@ static struct io_plan *io_tor_connect(struct io_conn *conn, struct reaching *rea
 	struct reaching_socks *reach_tor = tal(reach, struct reaching_socks);
 
 	reach_tor->port=htons(reach->addr.port);
-
-	port_addr = tal_fmt(tmpctx,"%u",reach->daemon->tor_proxy_port);
-	getaddrinfo((char *)reach->daemon->tor_proxy_ip, port_addr, NULL,&ai_tor);
+	port_addr = tal_fmt(tmpctx,"%u",reach->daemon->tor_proxyaddrs->port);
+	getaddrinfo((char *)fmt_wireaddr_without_port(tmpctx,reach->daemon->tor_proxyaddrs), port_addr, NULL,&ai_tor);
 	//getaddrinfo(tal_strdup(NULL,"127.0.0.1"),tal_strdup(NULL,"9050"), NULL,&ai_tor);
-	status_trace("torproyaddr_ip_port: %s %u",reach->daemon->tor_proxy_ip,reach->daemon->tor_proxy_port);
+	status_trace("torproyaddr_ip_port: %s",fmt_wireaddr_without_port(tmpctx,reach->daemon->tor_proxyaddrs));
```

Comment:
Would it not be better to use `fmt_wireaddr` instead of `fmt_wireaddr_without_port`?